### PR TITLE
[CFM-943] auto subscribe content authors.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-943

### CODE REVIEW

**Comment by anvmn** [comment]
@ordavidil 
This PR introduces 'user subscribe' model - waiting for Travis

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * @file
+ * Code for the C4M notifications subscriptions.
+ */
+
+/**
+ * Implements hook_node_insert().
+ *
+ * Flags inserted node with 'subscribe_c4m_follow_node' flag, for content that
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module]: fix docs

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ * Helper function.
+ *
+ * Returns an array of notification types names.
+ * These are the types that require notification.
+ *
+ * @return array
+ *   Notification types.
+ */
+function _c4m_user_subscribe_get_group_content_notification_types() {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module]: move to `c4m_og`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  global $user;
+
+  if (!in_array($flag_suffix, _c4m_user_subscribe_get_flags_suffixes())) {
+    // Unknown flag suffix.
+    return FALSE;
+  }
+
+  $wrapper = entity_metadata_wrapper('user', $user);
+  $field = "c4m_optout_{$flag_suffix}";
+  foreach ($wrapper->$field->value(
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module]: `foreach` should be in one line.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    return;
+  }
+
+  flag('flag', 'subscribe_c4m_follow_node', $node->nid);
+}
+
+/**
+ * Implements hook_node_update().
+ *
+ * Flags updated node with 'subscribe_c4m_follow_node' flag, for content that
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.module]: fix docs

**Comment by anvmn** [comment]
Move PR #1108 logic into c4m_user_subscribe module

File: .travis.yml [hunk 5]
```diff
@@ -1,5 +1,10 @@
 language: php
 
+branches:
```
Comment (ordavidil) [inline .travis.yml]: @anvmn 
This is wrong,
one of the reverts you've made should be revert again.

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.features.inc [hunk 6]
```diff
@@ -1,5 +1,4 @@
 <?php
-
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.features.inc]: @anvmn revert this one

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.features.inc]: use `[ci skip]` as a suffix for the commit message

**Comment by soniCaH** [comment]
As SA I get subscribed when authoring a new discussion in the "nobelprize" group.
But when I edit it as "alfrednobel" (the group owner), I don't get subscribed.

I get into this line of code:

{code}
      if (!$this->access($entity_id, $action, $account)) {
        $this->errors['access-denied'] = t('You are not allowed to flag, or unflag, this content.');
      }
{code}

so the user seems to have no access to flag (while he does have access to edit the node).

**Comment by ordavidil** [comment]
@anvmn 
please check the flag permissions.

**Pull request merged**